services/src/main/java/org/keycloak/services/resources/admin/AdminRoot.java

```diff
@@ -112,6 +119,11 @@ public Response masterRealmAdminConsoleRedirect() {
     @Path("index.{html:html}") // expression is actually "index.html" but this is a hack to get around jax-doclet bug
     @GET
     public Response masterRealmAdminConsoleRedirectHtml() {
+
+        if (!(Profile.isFeatureEnabled(Profile.Feature.ADMIN2) || Profile.isFeatureEnabled(Profile.Feature.ADMIN))) {
+            return Response.noContent().build();
```

Comment:
Should this be not-found rather than an empty response?

Reply:
Yep that makes more sense, I'll adapt the code.